.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish project package to PyPI
+
+on: push
```

Comment:
I guess we should have an alternative flow to publish the package ([example](https://github.com/actions/starter-workflows/blob/b2e55d25f26ccf467083ee31bb31aea8eb165641/ci/python-publish.yml))

```yaml
on:
  release:
    types: [created]
```

Reply:
Even better:

**Only tagged commits will be published to PyPI**
```yaml
- name: Publish package to PyPI
  if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
  uses: pypa/gh-action-pypi-publish@master
  with:
    password: ${{ secrets.PYPI_TOKEN }}
```

**Any other commit will be published to Test PyPI**
```yaml
- name: Publish package to TestPyPI
  uses: pypa/gh-action-pypi-publish@master
  with:
    password: ${{ secrets.PYPI_TOKEN }}
    repository_url: https://test.pypi.org/legacy/
```

[source](https://github.com/pypa/gh-action-pypi-publish)

Reply:
With that option, using the tags, it will release only when pushing tags, which doesn't include pushing code (only if we used an atomic push, but would be a tag to a branch and not to master). A better solution, using tags, would be a GitHub action reacting to tags (on.push.tag) and apply those tags after the acceptance of a pull request. For now, we will use the following strategy: at this moment every new commit on master is important to have it published in Pypi and so it will be releasing on every push, from every pull request; in the future, if there is a lot of activity we should change the strategy to group changes in single versions.